pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/ShuffleRewriteVisitor.java

```diff
@@ -150,19 +147,11 @@ public Set<Integer> visitMailboxSend(MailboxSendNode node, Void context) {
   @Override
   public Set<Integer> visitProject(ProjectNode node, Void context) {
     Set<Integer> oldPartitionKeys = node.getInputs().get(0).visit(this, context);
-
-    // all inputs carry over if they're still in the projection result
-    Set<Integer> partitionKeys = new HashSet<>();
-    for (int i = 0; i < node.getProjects().size(); i++) {
-      RexExpression rex = node.getProjects().get(i);
-      if (rex instanceof RexExpression.InputRef) {
-        if (oldPartitionKeys.contains(((RexExpression.InputRef) rex).getIndex())) {
-          partitionKeys.add(i);
-        }
-      }
+    if (rexExpressionListContainsAllPartitionKey(node.getProjects(), oldPartitionKeys)) {
```

Comment:
why is this necessary for project? projects shouldn't change partitioning, correct?

Reply:
project can make a partition become unpartitioned. for example
```
- Aggregation [ GroupSet = (0, 1, 2), AggCall = (...) ]
  - Project [ SelectList = [inputRef(0), inputRef(1), func(xx)], PartitionKeys = [] ]
    - MailboxReceived [ PartitionKeys = [inputRef(0), inputRef(1), inputRef(2)] ]
```
we need to remove the PartitionKeys (no longer inputRef [0, 1, 2]) because even the group set is now (0, 1, 2) it is no longer the same 0, 1, 2 it was referring to

Reply:
actually there's another bug. fixing